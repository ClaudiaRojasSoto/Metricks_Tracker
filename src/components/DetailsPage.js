import React from 'react';

const DetailsPage = () => (
  <h1>Details Page</h1>
);

export default DetailsPage;
