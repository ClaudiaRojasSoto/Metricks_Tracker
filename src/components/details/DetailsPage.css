.details-container {
  padding: 2em;
}

.details-header {
  display: flex;
  align-items: center;
  gap: 2em;
}

.country-flag {
  width: 100px;
  height: auto;
}

.details-main {
  display: flex;
  justify-content: space-around;
  align-items: start;
  margin-top: 2em;
}

.details-info {
  width: 45%;
}

.details-map {
  width: 45%;
  height: 400px; /* Altura del mapa */
  background: #e0e0e0; /* Color de fondo temporal hasta que se cargue el mapa */
}
