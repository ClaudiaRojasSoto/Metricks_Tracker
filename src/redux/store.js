import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './reducers/countriesReducer';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
