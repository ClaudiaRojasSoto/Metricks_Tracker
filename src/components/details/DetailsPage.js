import React from 'react';
import { useSelector } from 'react-redux'; // Importamos useSelector desde react-redux
import './DetailsPage.css';

const DetailsPage = () => {
  // Accedemos a los datos del paÃ­s desde el estado de Redux
  const countryData = useSelector((state) => state.countries.countryData);

  // Verificamos que countryData no sea null o undefined
  if (!countryData) {
    return <div>Loading...</div>;
  }

  // podemos desestructurar sus propiedades de manera segura
  const {
    name, capital, population, region, languages, area, flags, currencies,
  } = countryData;

  // Comprobaciones antes de acceder a las propiedades
  const commonName = name?.common ?? 'N/A';
  const capitalName = capital?.[0] ?? 'N/A';
  const populationCount = population ?? 'N/A';
  const regionName = region ?? 'N/A';
  const languageNames = languages ? Object.values(languages).join(', ') : 'N/A';
  const areaSize = area ?? 'N/A';
  const countryFlag = flags?.[0] ?? '';
  const countryCurrency = currencies ? Object.keys(currencies).join(', ') : 'N/A';

  return (
    <div className="details-container">
      <div className="details-header">
        <h1>
          {commonName}
        </h1>
        <img src={countryFlag} alt={`${commonName} Flag`} />
        <p>
          Capital:
          {capitalName}
        </p>
        <p>
          Population:
          {populationCount}
        </p>
        <p>
          Area:
          {areaSize}
        </p>
        <p>
          Region:
          {regionName}
        </p>
        <p>
          Languages:
          {languageNames}
        </p>
        <p>
          Currency:
          {countryCurrency}
        </p>
      </div>
    </div>
  );
};

export default DetailsPage;
