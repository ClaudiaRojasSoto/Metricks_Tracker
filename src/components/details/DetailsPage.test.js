import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import DetailsPage from './DetailsPage';

describe('DetailsPage tests', () => {
  let detailsPage;
  beforeAll(async () => {
    detailsPage = renderer.create(
      <Provider store={store}>
        <Router>
          <DetailsPage />
        </Router>
      </Provider>,
    ).toJSON();
  });

  it('Component should render correctly', () => {
    expect(detailsPage).toBeTruthy();
  });

  it('It should match its snapshot', () => {
    expect(detailsPage).toMatchSnapshot();
  });
});
