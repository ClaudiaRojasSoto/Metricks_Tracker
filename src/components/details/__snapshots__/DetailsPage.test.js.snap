// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage tests It should match its snapshot 1`] = `
<div>
  Loading...
</div>
`;
